libcaf_net/test/string_application.cpp

```diff
@@ -100,12 +100,11 @@ class string_application {
   template <class Parent>
   void write_message(Parent& parent,
                      std::unique_ptr<net::endpoint_manager::message> msg) {
-    std::vector<byte> buf;
     header_type header{static_cast<uint32_t>(msg->payload.size())};
-    buf.resize(sizeof(header_type));
-    memcpy(buf.data(), &header, buf.size());
-    buf.insert(buf.end(), msg->payload.begin(), msg->payload.end());
-    parent.write_packet(buf);
+    std::vector<byte> header_buf(sizeof(header_type));
+    std::vector<byte> payload(msg->payload.begin(), msg->payload.end());
+    memcpy(header_buf.data(), &header, header_buf.size());
+    parent.write_packet(make_span(header_buf), make_span(payload));
```

Comment:
I think `header_buf` is unnecessary:

```suggestion
    parent.write_packet(as_bytes(make_span(&header, 1)), make_span(payload));
```